packages/core/cache/src/FSCache.js

```diff
@@ -81,16 +83,44 @@ export class FSCache implements Cache {
     }
   }
 
+  #getFilePath(key: string, index: number): string {
+    return path.join(this.dir, `${key}-${index}`);
+  }
+
   hasLargeBlob(key: string): Promise<boolean> {
-    return this.fs.exists(this._getCachePath(`${key}-large`));
+    return this.fs.exists(this.#getFilePath(key, 0));
   }
 
-  getLargeBlob(key: string): Promise<Buffer> {
-    return this.fs.readFile(this._getCachePath(`${key}-large`));
+  async getLargeBlob(key: string): Promise<Buffer> {
+    const buffers: Promise<Buffer>[] = [];
+    for (let i = 0; await this.fs.exists(this.#getFilePath(key, i)); i += 1) {
+      const file: Promise<Buffer> = this.fs.readFile(this.#getFilePath(key, i));
+
+      buffers.push(file);
+    }
+
+    return Buffer.concat(await Promise.all(buffers));
```

Comment:
Will this not incur a `buffer.constants.MAX_LENGTH` violation? Or is the problem that `fs.readFile` has it's own, lower limit?

Reply:
Yeah, `readFile` is limited to 2GiB but Buffers are higher and dependent on the V8 version.

Reply:
ah, makes sense. I guess we're still gonna crash if we try to put together a 4GiB file, but (🤞) that's a new limit we won't hit any time soon.